File: Leetcode/Amazon spring 23 high frequency/k th factor of n/kth factor of n.cpp
class Solution {
public:
    int kthFactor(int n, int k) {
        vector<int> x;
        for(int i = 1; i<=n ; i++)
        {
            if(n%i == 0)
            x.push_back(i);
        }
        if(!x.empty() && k<=x.size())
        return x[k-1];
        return -1;
    }
};

File: Leetcode/Amazon spring 23 high frequency/k th factor of n/kth factor of n.py
class Solution:
    def kthFactor(self, n: int, k: int) -> int:
        factor = []
        i=1
        while i*i<=n:
            if n%i==0:
                factor.append(i)
                if i!=n//i:
                    factor.append(n//i)
            i+=1
        factor.sort()
        if k<=len(factor):
            return factor[k-1]
        return -1
